pkg/tenant/organizations.go

```diff
@@ -19,6 +19,64 @@ import (
 	"github.com/rotationalio/ensign/pkg/utils/ulids"
 )
 
+// OrganizationList fetches a list of organizations from Quarterdeck.
+//
+// Route: GET /v1/organization
+func (s *Server) OrganizationList(c *gin.Context) {
+	var (
+		ctx context.Context
+		err error
+	)
+
+	// User credentials are required to make the Quarterdeck request.
+	if ctx, err = middleware.ContextFromRequest(c); err != nil {
+		sentry.Error(c).Err(err).Msg("could not get user claims from authenticated request")
+		c.JSON(http.StatusUnauthorized, api.ErrorResponse(api.ErrInvalidUserClaims))
+		return
+	}
+
+	// Parse query parameters.
+	query := &api.PageQuery{}
+	if err = c.ShouldBindQuery(query); err != nil {
+		sentry.Warn(c).Err(err).Msg("could not parse page query request")
+		c.JSON(http.StatusBadRequest, api.ErrorResponse(api.ErrUnparsable))
+		return
+	}
+
+	// Build the Quarterdeck request.
+	req := &qd.OrganizationPageQuery{
+		PageSize:      int(query.PageSize),
+		NextPageToken: query.NextPageToken,
+	}
+
+	// Request a page of organizations from Quarterdeck.
+	var reply *qd.OrganizationList
+	if reply, err = s.quarterdeck.OrganizationList(ctx, req); err != nil {
+		sentry.Debug(c).Err(err).Msg("tracing quarterdeck error in tenant")
+		api.ReplyQuarterdeckError(c, err)
+		return
+	}
+
+	// Return page of organizations.
+	out := &api.OrganizationPage{
+		Organizations: make([]*api.Organization, 0),
+		NextPageToken: reply.NextPageToken,
+	}
+
+	for _, org := range reply.Organizations {
+		orgs := &api.Organization{
+			ID:        org.ID.String(),
+			Name:      org.Name,
+			Domain:    org.Domain,
+			Created:   db.TimeToString(org.Created),
+			LastLogin: db.TimeToString(org.LastLogin),
+		}
```

Comment:
Now that we are converting the DB model into an API model in two places, do you mind creating a `ToAPI` method in `pkg/tenant/db/organizations.go` similar to the one for the other resources? It wasn't necessary before but is now more important since we want to make sure the response is the same for both `OrganizationList` and `OrganizationDetail`.

While we're at it, we should also use this method in `OrganizationDetail` to make sure that we're populating `LastLogin` there as well.

Reply:
The only db model we have for `Organizations` is in Quarterdeck. We're not storing org details in trtl, so it feels like we adding one isn't necessary. Also, the Figma designs only display `Created` in `OrganizationDetail` and the `Organizations` table, so `LastLogin` would not be displayed anywhere, at least for now, if it's returned. If I recall, we're using `LastLogin` here to make it easier to return a user to the last org they were logged into.

Reply:
I see, we may want to have a helper method eventually but for the scope of this PR I think it's okay to go with what you have.